provisioning/miniprov/hare_mp/validator.py

```diff
@@ -0,0 +1,56 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import socket
+from hare_mp.store import ValueProvider
+
+
+class Validator:
+    def __init__(self, provider: ValueProvider):
+        super().__init__()
+        self.provider = provider
+
+    def _get_machine_id(self) -> str:
+        machine_id = self.provider.get_machine_id()
+        if not self.is_local_machine_id_valid(machine_id):
+            raise RuntimeError(f'invalid machine id {machine_id}')
+        return machine_id
+
+    def is_first_node_in_storage_set(self) -> bool:
+        machine_id = self._get_machine_id()
+        cluster_id = self.provider.get_cluster_id()
+        storage_set_id = self.provider.get_storage_set_id()
+        server_nodes_key: str = (f'cluster>{cluster_id}>'
+                                 f'storage_set[{storage_set_id}]>server_nodes')
+        server_nodes = self.provider.get(server_nodes_key)
+        return server_nodes[0] == machine_id
```

Comment:
machine_id will be in format '0a31807456be14965f31b49314f5664e'
whereas 'server_nodes' will be in format as shown below,
cluster>{cluster-id}>storage_set[N]>server_nodes = ["srvmachine-1", "srvmachine-2", "srvmachine-3"] (taken from sheet).
@mssawant Please check if they are really in same format.

Reply:
Per my understanding server_nodes contains machine id's. In other words, I believe Mandar's patch is correct.

Reply:
If you read the description of the `cluster>{cluster-id}>storage_set[N]>server_nodes` key it says its a list of machine-ids.
```
List of server node machine-ids belonging to Storage-Set N
```
We have not hardcoded anything, we will use whatever we get from the conf-store.

Reply:
You are correct,
Sample json generated by Nitin contains following lines,
"storage_set": [
        {
          "name": "storage-set-1",
          "server_nodes": [
            "a971a87aace34e05b4747bb3376d6661"
          ],